process/sync/shardblock.go

```diff
@@ -377,9 +498,28 @@ func (boot *ShardBootstrap) SyncBlock() error {
 		return err
 	}
 
+	defer func() {
+		if err != nil {
```

Comment:
move it into a function. too long for defer

Reply:
Done